code/modules/chemistry/Chemistry-Tools.dm

```diff
@@ -919,6 +927,88 @@ proc/ui_describe_reagents(atom/A)
 			src.connected_containers.Add(container)
 
 
+/obj/item/reagent_containers/glass/plumbing/dropper
+	name = "dropper funnel"
+	desc = "A glass tube for adding reagents to containers at a customisable rate. Can be hooked up to many types of containers."
+	icon = 'icons/obj/items/chemistry_glassware.dmi'
+	icon_state = "dropper_funnel_off"
+	splash_all_contents = FALSE
+	object_flags = OPENCONTAINER | SUPPRESSATTACK
+	initial_volume = 100
+	accepts_lid = TRUE
+	fluid_overlay_states = 8
+	container_style = "dropper_funnel"
+	var/open = FALSE
+	var/flow_rate = 1
+	HELP_MESSAGE_OVERRIDE({"Click and drag this onto other glassware to connect it.
+	Flow rate can be adjusted either in your hand, or with a screwdriver.
+	To pick this up, click and drag it onto your person."})
+
+	disposing()
+		src.remove_container()
+		. = ..()
+
+	mouse_drop(atom/over_object, src_location, over_location)
+		if (ismob(over_object))
+			var/mob/user = over_object
+			if (user == usr && !user.restrained() && !user.stat && (user.contents.Find(src) || in_interact_range(src, user)))
+				if (!user.put_in_hand(src))
+					return ..()
+		else
+			..()
+
+	add_line(var/obj/container)
+		var/datum/lineResult/result = drawLine(src, container, "condenser", "condenser_end", src.pixel_x-1, src.pixel_y-13, container.pixel_x, container.pixel_y + container.get_chemical_effect_position())
+		result.lineImage.pixel_x = -src.pixel_x
+		result.lineImage.pixel_y = -src.pixel_y
+		result.lineImage.layer = src.layer+0.01
+		src.UpdateOverlays(result.lineImage, "tube\ref[container]")
+	attack_self(mob/user as mob)
+		if(current_lid)
+			boutput(user, SPAN_NOTICE("You pop the lid off of the [src]."))
+			remove_current_lid(user)
```

Comment:
This supports lids but appears to be missing the icon state required to show them?

Reply:
d'oh. knew i forgor something